Makefile

```diff
@@ -1,3 +1,5 @@
+include hack/makefile-vars.mk
```

Comment:
This is a refactor. Keep it in a separate PR.

Reply:
Umnaga This is a small change I think we can keep it here in the same PR. Maybe separate commit if you insist.

Reply:
This is kind of a refactor PR only, This is a new way of building a catalog and I want to start with using the makefile vars & move away as much as possible from the common.sh format. And I have kept this in a separate commit so I think this can go here.